### PR TITLE
Rt2.12 static route redistribution

### PR DESCRIPTION
**Pull request by mastarkey**
This test is for RT21.2 

This code is mostly complete however it is missing the portion of route validation on the OTG

### CODE REVIEW

**Comment by octpetre** [comment]
Please move this under _**feature/isis/otg_tests**_

File: feature/isis/otg_tests/static_route_isis_redistribution_test/static_route_isis_redistribution_test.go
Comment (Ankur19) [inline feature/isis/otg_tests/static_route_isis_redistribution_test/static_route_isis_redistribution_test.go:484]: Fix

File: feature/isis/otg_tests/static_route_isis_redistribution_test/static_route_isis_redistribution_test.go
Comment (Ankur19) [inline feature/isis/otg_tests/static_route_isis_redistribution_test/static_route_isis_redistribution_test.go:710]: Uncomment and add t.Skip()

File: feature/isis/otg_tests/static_route_isis_redistribution_test/static_route_isis_redistribution_test.go
Comment (mastarkey) [reply feature/isis/otg_tests/static_route_isis_redistribution_test/static_route_isis_redistribution_test.go:160]: This is a temporary solution as Ondatra uses subscribe to check this path however currently this being resolved internally to fix, so needed to do a get to work around this for time being.

File: feature/isis/otg_tests/static_route_isis_redistribution_test/static_route_isis_redistribution_test.go
Comment (mastarkey) [reply feature/isis/otg_tests/static_route_isis_redistribution_test/static_route_isis_redistribution_test.go:131]: Tried to integrate this but when i use this it keeps returning negative integer vs directly getting the flow name counters

File: feature/isis/otg_tests/static_route_isis_redistribution_test/static_route_isis_redistribution_test.go
Comment (Ankur19) [inline feature/isis/otg_tests/static_route_isis_redistribution_test/static_route_isis_redistribution_test.go:157]: You can use ygnmi ops to use Get instead of Subscribe:

`gnmi.Get(t, dut.GNMIOpts().WithYGNMIOpts(ygnmi.WithUseGet())), .........)`

We would also need to update the hardcoded `DEFAULT` network instance with `deviations.DefaultNetworkInstance(dut)`

File: feature/isis/otg_tests/static_route_isis_redistribution_test/static_route_isis_redistribution_test.go
Comment (Ankur19) [reply feature/isis/otg_tests/static_route_isis_redistribution_test/static_route_isis_redistribution_test.go:128]: We can use `otgutils.GetFlowStats()` if the loss percentage is a problem.

File: feature/isis/otg_tests/static_route_isis_redistribution_test/static_route_isis_redistribution_test.go
Comment (mastarkey) [reply feature/isis/otg_tests/static_route_isis_redistribution_test/static_route_isis_redistribution_test.go:157]: tried to implement this but had hit the following errors. not sure if this needs to be structured in a different way.

path := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).TableConnection(
						oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_STATIC,
						oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_ISIS,
						oc.Types_ADDRESS_FAMILY_IPV4).State()

output := gnmi.Get(t, dut.GNMIOpts().WithYGNMIOpts(ygnmi.WithUseGet(),
						ygnmi.WithEncoding(gpb.Encoding_JSON_IETF)), path)
					t.Log(output)
					
noncompliant data encountered while unmarshalling: Noncompliance Errors by category (see https://github.com/openconfig/ygnmi#noncompliance-errors):
Path Noncompliance Errors: None
Type Noncompliance Errors:

&{DUT %!s(func(context.Context) (gnmi.GNMIClient, error)=0x10a02e7a0) map[] %!s(bool=true) <nil> [%!s(ygnmi.Option=0x106fa8ab0) %!s(ygnmi.Option=0x106fa8470)]} at /network-instances/network-instance[name=DEFAULT]/table-connections/table-connection[address-family=IPV4][dst-protocol=ISIS][src-protocol=STATIC]: path origin:"openconfig"  elem:{name:"network-instances"}  elem:{name:"network-instance"  key:{key:"name"  value:"DEFAULT"}}  elem:{name:"table-connections"}  elem:{name:"table-connection"  key:{key:"address-family"  value:"IPV4"}  key:{key:"dst-protocol"  value:"ISIS"}  key:{key:"src-protocol"  value:"STATIC"}}: value not present

File: feature/isis/otg_tests/static_route_isis_redistribution_test/static_route_isis_redistribution_test.go
Comment (mastarkey) [reply feature/isis/otg_tests/static_route_isis_redistribution_test/static_route_isis_redistribution_test.go:128]: was able to integrate this [otgutils.GetFlowLossPercent()](https://github.com/openconfig/featureprofiles/blob/main/internal/otgutils/actions.go#L54) not sure why it worked this time maybe something flaky in my setup at the time.

File: feature/isis/otg_tests/static_route_isis_redistribution_test/static_route_isis_redistribution_test.go
Comment (Ankur19) [reply feature/isis/otg_tests/static_route_isis_redistribution_test/static_route_isis_redistribution_test.go:136]: This is right. 

`https://github.com/openconfig/ygnmi#noncompliance-errors` non compliance errors are for logging only. The real issue here is `value not present`.

I see that in your previous implementation you used the `Config` path, however, in the new one you used `State`. The state value is not present and hence the error.

Comment (Ankur19) [reply feature/isis/otg_tests/static_route_isis_redistribution_test/static_route_isis_redistribution_test.go:136]: Use the `Config` path instead?

Comment (mastarkey) [reply feature/isis/otg_tests/static_route_isis_redistribution_test/static_route_isis_redistribution_test.go:136]: The problem with the `config` path is currently its not returning the `disable-metric-propagation` value so i would still end up needing to do this with the same approach as i have already implemented.

**Comment by mastarkey** [comment]
Are there any pending changes if not can we merge this ?

**Comment by Ankur19** [comment]
We can merge this once the merge conflicts are resolved. 

Thank you for your contribution!

**Comment by mastarkey** [comment]
@Ankur19 conflicts are resolved

**Review by Ankur19 — APPROVED** [review]

**Pull request merged**
